File: gera-pagamentos/src/main/java/br/com/alfser/projeto/pagamentos/services/BrokerProducerService.java
package br.com.alfser.projeto.pagamentos.services;

import br.com.alfser.projeto.pagamentos.common.BrokerTopic;
import br.com.alfser.projeto.pagamentos.dtos.PagamentoUpdateStatusDTO;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
public class BrokerProducerService {
    private final KafkaTemplate<String, String> kafkaTemplate;
    private final ObjectMapper objectMapper;

    public BrokerProducerService(KafkaTemplate<String, String> kafkaTemplate, ObjectMapper objectMapper) {
        this.kafkaTemplate = kafkaTemplate;
        this.objectMapper = objectMapper;
    }

    public void sendMessage(PagamentoUpdateStatusDTO dto) throws JsonProcessingException {
        String json = objectMapper.writeValueAsString(dto);
        System.out.println("PRODUCE MESSAGE: "+json);
        kafkaTemplate.send(BrokerTopic.PAGAMENTO_PENDENTE, json);
    }
}

File: processa-pagamentos/src/main/kotlin/br/com/alfser/projeto/processa_pagamentos/services/BrokerProducerService.kt
package br.com.alfser.projeto.processa_pagamentos.services

import br.com.alfser.projeto.processa_pagamentos.common.BrokerTopic
import br.com.alfser.projeto.processa_pagamentos.dtos.PagamentoDTO
import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service

@Service
class BrokerProducerService (
    private val kafkaTemplate: KafkaTemplate<String?, String?>,
    private val objectMapper: ObjectMapper
){

    @Throws(JsonProcessingException::class)
    fun sendMessage(dto: PagamentoDTO?) {
        val json = objectMapper.writeValueAsString(dto)
        println("PRODUCE MESSAGE: " + json)
        kafkaTemplate.send(BrokerTopic.PAGAMENTO_STATUS, json)
    }
}